File: Recorder/src/main/java/com/github/ezauton/recorder/SequentialDataFrame.java
package com.github.ezauton.recorder;

import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * Describes a recording frame about time-based data
 * <br>
 * Essentially a recording for one instant
 */
public abstract class SequentialDataFrame
{
    @JsonProperty
    protected double time;

    protected SequentialDataFrame(double time) {this.time = time;}

    protected SequentialDataFrame() {}

    public double getTime()
    {
        return time;
    }
}

File: Recorder/src/main/java/com/github/ezauton/recorder/SequentialDataFrame.kt
package com.github.ezauton.recorder

import com.fasterxml.jackson.annotation.JsonProperty


/**
 * Describes a recording frame about time-based data
 * <br></br>
 * Essentially a recording for one instant
 */
abstract class SequentialDataFrame {
  @JsonProperty
  var time = 0.0
    protected set

  protected constructor(time: Double) {
    this.time = time
  }

  protected constructor() {}
}
